Funcky/Extensions/QueueExtensions.cs

```diff
@@ -0,0 +1,37 @@
+using System.Collections.Concurrent;
+
+namespace Funcky.Extensions
+{
+    public static partial class QueueExtensions
+    {
+#if QUEUE_TRY_OVERLOADS
+        [Pure]
+        public static Option<TItem> DequeueOrNone<TItem>(this Queue<TItem> queue)
+            where TItem : notnull
+            => queue.TryDequeue(out var result)
+                ? result
+                : Option<TItem>.None();
+
+        [Pure]
+        public static Option<TItem> PeekOrNone<TItem>(this Queue<TItem> queue)
+            where TItem : notnull
+            => queue.TryPeek(out var result)
+                ? result
+                : Option<TItem>.None();
+#endif
```

Comment:
I think we should still support these overloads for .NET Standard 2.0.
We can use the regular methods and handle the `InvalidOperationException`. What do you think?

Reply:
Good idea, used your helper in 72f285f92b0cc54942b08db104a6276af40b4383
The helper has been extracted to it's own class, maybe we want to promote it to public at some point